Client/mods/deathmatch/logic/luadefs/CLuaAudioDefs.cpp

```diff
@@ -153,16 +153,24 @@ int CLuaAudioDefs::PlaySound(lua_State* luaVM)
             {
                 SString strFilename;
                 bool    bIsURL = false;
+                bool    bIsRawData = false;
                 if (CResourceManager::ParseResourcePathInput(strSound, pResource, &strFilename))
                     strSound = strFilename;
                 else
-                    bIsURL = true;
+                {
+                    SString signature = strSound.SubStr(0, 4).ToLower();
+                    if ((memcmp(signature, "http", 4) == 0 || memcmp(signature, "ftp", 3) == 0)
+                        && (strSound.length() <= 2048 || strSound.find('\n') == SString::npos))
+                        bIsURL = true;
```

Comment:
Hmmm so `httplmao` is considered as a valid url ?
From what I see, I assume we were already relying on the internals of **PlaySound** to have the last word on the validity of the input. Do we consider that check as "good enough" ? (I can't see it producing crashes from what the previous code looked like, I'm fine with that then)

Reply:
> Hmmm so `httplmao` is considered as a valid url ?

Just a string that seems good enough to pass to BASS as a URL, without it exploding. After all, it seems reasonable that, as you said, BASS has the last word on its validity. Duplicating the checks that BASS already does is a waste of time.